lua/lspconfig/server_configurations/foam.lua

```diff
@@ -0,0 +1,33 @@
+local util = require("lspconfig.util")
+local bin_name = "foam-ls"
+local cmd = { bin_name, "--stdio" }
+
+if vim.fn.has("win32") == 1 then
+	cmd = { "cmd.exe", "/C", bin_name, "--stdio" }
+end
+
+return {
+	default_config = {
+		cmd = cmd,
+		filetypes = { "foam", "OpenFOAM" },
+		root_dir = function(fname)
+			return util.search_ancestors(fname, function(path)
+				if util.path.exists(util.path.join(path, "system", "controlDict")) then
+					return path
+				end
+			end)
+		end,
+		log_level = vim.lsp.protocol.MessageType.Warning,
```

Comment:
Why do you need to set a default log level?

Reply:
An artifact from early developing phases; Thanks for catching that!